### PR TITLE
Generate an Aeon request link using a web-hosted EAD file

### LINKED ISSUE #643: Generate a request link for Aeon using a web-hosted EAD file
Requires #641. Supports Indiana, and Michigan-Special Collections.

**Acceptance criteria**

Given a repository with a configured Aeon instance:

* Add the ability to generate a link to a web-hosted EAD file. (This could be created added in a new field at index time, or generated from existing index field data.) In the variable below, this will be referred to as `ead_url`
* Have the request button/link point to the following: `"#{aeon_web_url}?Action=10&Form=31&Value=#{ead_url}"`
  * For example, given an `aeon_web_url` of `"https://iris.lib.umich.edu/aeon/aeon.dll"` and an `ead_url` of `"https://quod.lib.umich.edu/s/sclead/eads/hennacy.xml"`, the link should point to `"https://iris.lib.umich.edu/aeon/aeon.dll?Action=10&Form=31&Value=https://quod.lib.umich.edu/s/sclead/eads/hennacy.xml"`

### PR DESCRIPTION
**Pull request by camillevilla**
Closes #643
Paired w/ @mejackreed on this.

This PR:
- makes the parent's `unitid` available in the document
- adds a method for rendering request button(s) based off a repository's configured request types
- adds a class for handling Aeon requests w/ a web-hosted EAD 
Sample fixture: 
http://localhost:3000/catalog/umich-bhl-2016071

### CODE REVIEW

File: spec/fixtures/config/repositories.yml [hunk 1]
```diff
@@ -62,3 +62,7 @@ umich-bhl:
   phone: ''
   contact_info: 'bentley.ref@umich.edu'
   thumbnail_url: 'https://upload.wikimedia.org/wikipedia/commons/thumb/9/93/Bhlexterior.jpg/150px-Bhlexterior.jpg'
+  request_types:
+    aeon_web_ead:
+      request_url: 'https://sample.request.com'
+      request_mappings: 'Action=10&Form=31&Value=ead_url'
```
Comment (camillevilla) [inline spec/fixtures/config/repositories.yml]: The `ead_url` here invokes the `ead_url` method in the `AeonWebEad` class

**Comment by seanaery** [comment]
Hi @camillevilla -- I pulled down this branch and generated a clean test app with it but am unable to see a request button get rendered. Not sure what I might be missing.

Here's what happens:
1. Visit http://localhost:3000/catalog/umich-bhl-2016071
2. The Contents tab remains at the unclickable "No Content Inventory"

The async request to http://localhost:3000/catalog?f%5Bcomponent_level_isim%5D%5B%5D=1&f%5Bcollection_sim%5D%5B%5D=Frank+Henry+Schultz+photograph+collection%2C+Circa+1909-1919&view=hierarchy returns a `500`. Errors [on this line](https://github.com/projectblacklight/arclight/blob/643-web-ead/app/views/arclight/requests/_aeon_web_ead.html.erb#L3) when hitting that URL. Message as follows:

Screen Shot 2019-09-18 at 10 31 59 AM

**Comment by estelendur** [comment]
I was not able to replicate the error @seanaery got.

**Comment by camillevilla** [comment]
Update: rebasing and working on moving fixtures over to sul-spec so it doesn't collide with #792.

**Comment by seanaery** [comment]
Hi @camillevilla -- it's working for me now... and it looks really great! Not sure what I was doing incorrectly before, but please disregard the error reporting above :-)

**Comment by seanaery** [comment]
Hi @camillevilla -- quick check on this PR. I can see those links rendering in the Contents tab on http://localhost:3000/catalog/m0198-xml and Collection Contents section on a component e.g. http://localhost:3000/catalog/m0198-xmlaspace_ref11_d0s though the gray Request button in the upper-right on the component page doesn't yet lead to the Aeon request link. Is that ticketed elsewhere?

**Pull request merged**